projects/plugins/boost/app/modules/critical-css/class-critical-css.php

```diff
@@ -652,31 +654,76 @@ public function get_critical_css() {
 	 * @see style_loader_tag
 	 */
 	public function asynchronize_stylesheets( $html, $handle, $href, $media ) {
+		// If is AMP, do not alter the stylesheet loading.
 		if ( function_exists( 'is_amp_endpoint' ) && is_amp_endpoint() ) {
 			return $html;
 		}
 
+		// If there is no critical CSS, do not alter the stylesheet loading.
 		if ( false === $this->get_critical_css() ) {
 			return $html;
 		}
 
-		if ( ! apply_filters( 'jetpack_boost_async_style', true, $handle ) ) {
+		$available_methods = array(
+			'instant'  => 'media="' . $media . '"',
```

Comment:
Hm. This is an interesting approach. I like how the code reads below when this is used:

```
if ( ! isset( $available_methods[ $method ] ) || 'instant' === $method  ) {
  return $html
}
```

But right here `'media="' . $media . '"'` doesn't seem to make much sense because it's actually never used, as we're only using the `instant` key and not the value. In other words, if `instant` is present, we return the original HTML, which we should I think. It's just that the code here suggests that we might be modifying the media value, although we're actually not.

I think this would just be simpler if we only provided the 2 available methods that Boost is going to apply. And set `async` to be the default (like you've done below) and then if anything other than one of our methods is returned - we automatically just quit filtering and return the original HTML.

That way you can also cancel Boost's filter easily by adding any of these to your code:


```php
add_filter( 'jetpack_boost_async_style', '__return_false' );
add_filter( 'jetpack_boost_async_style', '__return_null' );
add_filter( 'jetpack_boost_async_style', '__return_empty_string' );
// etc.
```

Reply:
I agree that `'media="' . $media . '"'` is unused. The main disadvantage of not adding that is - users will never know how to avoid Boost's filter by looking at the `$available_methods`. I think what we need here docblock with an example as you suggested. Adding that!